src/Jackalope/Version/VersionHandler.php

```diff
@@ -245,5 +247,172 @@ private function createFrozenNode(NodeInterface $versionNode, NodeInterface $nod
 
             $frozenNode->setProperty($propertyName, $property->getValue());
         }
+
+        foreach ($node->getNodes() as $childNode) {
+            /** @var NodeInterface $childNode */
+            // TODO also use the commented lines when getDefinition is implemented
+            //$onParentVersion = $childNode->getDefinition()->getOnParentVersion();
```

Comment:
yes, i generalized that and created https://github.com/jackalope/jackalope/pull/292

please try if that works for you and give feedback. once its merged, you could rebase the versioning jackalope branch onto master and this PR onto your versioning branch, then you have the feature everywhere.